src/vm/arm64/asmhelpers.asm

```diff
@@ -1326,5 +1329,62 @@ Fail
     LEAF_END
 #endif
 
+;
+; JIT Static access helpers when coreclr host specifies single appdomain flag 
+;
+
+; ------------------------------------------------------------------
+; void* JIT_GetSharedNonGCStaticBase(SIZE_T moduleDomainID, DWORD dwClassDomainID)
+
+    LEAF_ENTRY JIT_GetSharedNonGCStaticBase_SingleAppDomain
+    ; If class is not initialized, bail to C++ helper
+    add x2, x0, #DomainLocalModule__m_pDataBlob
+    ldrb w2, [x2, w1]
```

Comment:
Hmm this is a modification in the format compared to the document i have and may be we are using a bit old assembler. Will need to check.
I actually wanted this instruction
`ldrb w2, [x2, w1, UXTW]`

Second parameter to the function is a dword so it is 32-bits and since it is an ID we should xero extend it to 64-bits.